File: core/src/main/java/nl/t64/game/rpg/Engine.java
package nl.t64.game.rpg;

import com.badlogic.gdx.Game;
import com.badlogic.gdx.Gdx;
import nl.t64.game.rpg.constants.ScreenType;
import nl.t64.game.rpg.profile.ProfileManager;
import nl.t64.game.rpg.screens.ScreenManager;
import nl.t64.game.rpg.screens.world.MapManager;


public class Engine extends Game {

    public static float runTime = 0f;

    final ResourceManager resourceManager;
    final Settings settings;
    final ProfileManager profileManager;
    final GameData gameData;
    final ScreenManager screenManager;
    final MapManager mapManager;

    public Engine(Settings settings) {
        this.resourceManager = new ResourceManager();
        this.settings = settings;
        this.profileManager = new ProfileManager();
        this.gameData = new GameData();
        this.screenManager = new ScreenManager();
        this.mapManager = new MapManager();
    }

    private static void updateRunTime() {
        runTime += Gdx.graphics.getDeltaTime();
    }

    @Override
    public void create() {
        profileManager.addObserver(gameData);
        profileManager.addObserver(mapManager);

        screenManager.setScreen(ScreenType.MENU_MAIN);
    }

    @Override
    public void render() {
        updateRunTime();
        super.render();
    }

    @Override
    public void dispose() {
        screenManager.dispose();
    }

}

File: core/src/nl/t64/game/rpg/Engine.kt
package nl.t64.game.rpg

import com.badlogic.gdx.Game
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.controllers.Controllers
import nl.t64.game.rpg.audio.AudioManager
import nl.t64.game.rpg.constants.ScreenType
import nl.t64.game.rpg.screens.ScreenManager
import nl.t64.game.rpg.screens.world.MapManager


open class Engine : Game() {

    companion object {
        private var runTime = 0f

        private fun updateRunTime() {
            runTime += Gdx.graphics.deltaTime
        }

        fun getRunTime(): Float {
            return runTime
        }
    }

    lateinit var preferenceManager: PreferenceManager
    val profileManager: ProfileManager = ProfileManager()
    val resourceManager: ResourceManager = ResourceManager()
    val gameData: GameData = GameData()
    val screenManager: ScreenManager = ScreenManager()
    val audioManager: AudioManager = AudioManager()
    val mapManager: MapManager = MapManager()
    val brokerManager: BrokerManager = BrokerManager()
    val gamepadMapping: GamepadMapping = GamepadMapping()

    override fun create() {
        preferenceManager = PreferenceManager()
        preferenceManager.setFullscreenAccordingToPreference()

        brokerManager.profileObservers.addObserver(gameData)
        brokerManager.profileObservers.addObserver(mapManager)

        Controllers.addListener(gamepadMapping.controllerToInputAdapter)

        screenManager.setScreen(ScreenType.MENU_MAIN)
    }

    override fun render() {
        updateRunTime()
        super.render()
    }

    override fun dispose() {
        screenManager.dispose()
    }

}
